src/util/heap/monotonepageresource.rs

```diff
@@ -294,6 +294,33 @@ impl<VM: VMBinding> MonotonePageResource<VM> {
         }
     }
 
+    /// Iterate through allocated regions, and invoke the given function for each region.
+    pub fn for_allocated_regions<F>(&self, f: F)
+    where
+        F: Fn(Address, usize),
```

Comment:
While `Fn` works for now, I recommend changing this to `FnMut` so that the user of `for_allocated_region` can have mutable states in the `f` closure.

Reply:
Done.